### DISCUSSION TITLE
How to use /etc/hosts file effectively?

### DISCUSSION THREAD

**Discussion by birds-swim** [SpiralLinux/SpiralLinux-project#467] [Q&A]
I've had trouble in the past getting my system to use my `/etc/hosts` file.

Is there some shake and dance going on between NetworkManager and systemd that I don't understand? 

I like setting up my Windows and Linux machines to use both a hosts file and NextDNS as the main DNS resolver. I usually put only malicious domains in my hosts file to guarantee my computer will never even connect to those bad websites full of malware.

I don't know if it's a stupid plan, but it sounds cool to me. Hosts protects me from malware (as a layer of defense) and then NextDNS filters out any advertisements/tracking domains. Then uBlock Origin does the rest in my browser.

**Comment by ghost** [comment]
I think you misunderstand the /etc/host use; you should freshen up on the relevant RFC about how name resolution works.  Windows does NOT honor proper RFC intent btw  :(

The "local" host file is consulted first for name resoltion prior to DNS resolition so unless you have used a firewall filter to mangle the malware spammers and such I am not sure how protected you are?

Maybe I misunderstood your question?  You use the host file to stop YOU from accessing Internet sites by looping any NAMED domain to 127.0.0.1 but the remote domain can always see you  : P

Use a firewall and not the host file   ... 


Just my thoughts,
Bill
